### PR TITLE
feat: add Lit to UIFrameworkTabs

### LINKED ISSUE #3870: [CONTENT] - add nanostores installation and usage for Lit
### 📚 Subject area/topic

nano-stores, state management

### 📋 Page(s) affected (or suggested, for new content)

https://docs.astro.build/en/core-concepts/sharing-state/#why-nano-stores

### 📋 General description or bullet points (if proposing new content)

- Add example for installing `@nanostores/lit`, alongside `@nanostores/preact`, `@nanostores/react`, `@nanostores/solid` and `@nanostores/vue` in [this section](https://docs.astro.build/en/core-concepts/sharing-state/#installing-nano-stores)

image

- Add example for using `@nanostores/lit` alongside `@nanostores/preact`, `@nanostores/react`, `@nanostores/solid` and `@nanostores/vue` in [this section](https://docs.astro.build/en/core-concepts/sharing-state/#usage-example---ecommerce-cart-flyout)

image

### 🖥️ Reproduction in StackBlitz (if reporting incorrect content or code samples)

-

### PR DESCRIPTION
**Pull request by ElianCodes**
#### What kind of changes does this PR include?

- Minor content fixes (broken links, typos, etc.)
- New or updated content
- Changes to the docs site code

#### Description

- Closes #3870 
- Adds Lit to `UIFrameworkTabs` at latest position
- update `Tabs.tsx` component to hide UI Framework tabs when panel section / slot is empty

Example page: https://deploy-preview-4150--astro-docs-2.netlify.app/en/core-concepts/sharing-state/#installing-nano-stores

TODO: 
- [x] fix Lit showing up twice (see image below)
- [x] Add examples for Lit to Nanostores pages

Screenshot 2023-08-09 at 17 57 49

### CODE REVIEW

**Comment by ElianCodes** [comment]
I'm still working on this PR, but @delucis, I definitely need your help / review.

Right now, Lit automatically gets displayed first in the `UIFrameworkTabs` component, but I'd rather show it last, but a bit stuck on that.

Screenshot 2023-08-09 at 14 53 17

Next to that, there is a typing issue for the `isPanelEmpty` function in `Tabs.tsx`, which makes sense, but I'm missing out on how I can detect better if a panel slot is empty!

```ts
function isPanelEmpty(key: TabSlot): boolean {
	const panel = panels.find(
		([panelKey]) => panelKey === `${panelSlotKey}${getBaseKeyFromTab(key)}`
	);
	if (!panel || panel[1]?.props.value === '') {
		return true;
	}
	return false;
}
```

Much appreciate your help!

File: src/components/tabs/UIFrameworkTabs.astro [hunk 1]
```diff
@@ -3,15 +3,17 @@ import Tabs from './Tabs';
...
-	<Fragment slot="tab.react">React</Fragment>
-	<Fragment slot="tab.solid">Solid</Fragment>
-	<Fragment slot="tab.svelte">Svelte</Fragment>
-	<Fragment slot="tab.vue">Vue</Fragment>
+  <Fragment slot="tab.preact">Preact</Fragment>
+  <Fragment slot="tab.react">React</Fragment>
+  <Fragment slot="tab.solid">Solid</Fragment>
+  <Fragment slot="tab.svelte">Svelte</Fragment>
+  <Fragment slot="tab.vue">Vue</Fragment>
+  <Fragment slot="tab.z_lit">Lit</Fragment>
```
Comment (TheOtterlord) [inline src/components/tabs/UIFrameworkTabs.astro:11]:
```suggestion
  <Fragment slot="tab.zLit">Lit</Fragment>
```

File: src/components/tabs/UIFrameworkTabs.astro [hunk 2]
```diff
@@ -3,15 +3,17 @@ import Tabs from './Tabs';
...
-	<Fragment slot="panel.react"><slot name="react" /></Fragment>
-	<Fragment slot="panel.solid"><slot name="solid" /></Fragment>
-	<Fragment slot="panel.svelte"><slot name="svelte" /></Fragment>
-	<Fragment slot="panel.vue"><slot name="vue" /></Fragment>
+  <Fragment slot="panel.preact"><slot name="preact" /></Fragment>
+  <Fragment slot="panel.react"><slot name="react" /></Fragment>
+  <Fragment slot="panel.solid"><slot name="solid" /></Fragment>
+  <Fragment slot="panel.svelte"><slot name="svelte" /></Fragment>
+  <Fragment slot="panel.vue"><slot name="vue" /></Fragment>
+  <Fragment slot="panel.z_lit"><slot name="lit" /></Fragment>
```
Comment (TheOtterlord) [inline src/components/tabs/UIFrameworkTabs.astro:18]:
```suggestion
  <Fragment slot="panel.zLit"><slot name="lit" /></Fragment>
```

**Review by TheOtterlord — COMMENTED** [review]
These two changes fix the problem causing Lit to be displayed at the beginning and end of a tabs list

File: src/components/tabs/UIFrameworkTabs.astro (hunk 1 above)
Comment (ElianCodes) [reply src/components/tabs/UIFrameworkTabs.astro:11]: I was just looking into that lol!
Thanks!

**Review by delucis — COMMENTED** [review]
Thanks for tackling this @ElianCodes! I’m not sure we want to add the is panel empty support as it doesn’t work well with our current tab-syncing behaviour.

For example, if I switch to Lit in the install instructions:

image

We get this weird behaviour for tab sets that are missing an Lit entry:

image

I guess we need to fill in those later examples.

**Comment by ElianCodes** [comment]
Yeah, I noticed that too, but my initial idea was to add all Lit examples to the Nanostores page, that way, we won't have the panel selection problem on that page.

Kinda like write all examples, or none. Do you think that's acceptable? Or would you rather have me adding Lit examples or placeholders wherever `UIFrameworkTabs` is used?

I'll leave this to you @delucis (also cc'ing @sarah11918 into the discussion)

**Pull request merged**